Fix small typos in the guides [skip ci] (#40312)
rails/rails: guides/source/active_support_instrumentation.md

end
```

> You may also pass a block with only one argument, it will yield an event object to the block:

Reviewer: Since we're touching this line, would you mind if we also fix the run-on sentence? :pray: 

```suggestion
You may also pass a block that accepts only one argument, and it will receive an event object:
```

Author: Yes, I found that sentence a bit clucky as well. 😄 
Would you mind if we make it even simpler?
```suggestion
You may also pass a block, which receives an event object:
```

Author: The previous paragraph mentions 'multiple arguments':
> Defining all those block arguments each time can be tedious. You can easily create an ActiveSupport::Notifications::Event from block arguments like this:

So I've changed it to @jonathanhefner 's suggestion.
